JNWCollectionView/JNWCollectionViewFramework.m

```diff
@@ -131,9 +137,23 @@ - (id)initWithCoder:(NSCoder *)aDecoder {
 	return self;
 }
 
+#pragma mark Notifications
+
+- (void)preferredScrollerStyleDidChangeNotification:(NSNotification *)note {
+	NSView *documentView = self.documentView;
+	
+	if (!CGSizeEqualToSize(NSSizeToCGSize(documentView.frame.size), NSSizeToCGSize(self.contentSize))) {
+		CGRect documentFrame = documentView.frame;
+		documentFrame.size.width = self.contentSize.width;
+		documentView.frame = documentFrame;
```

Comment:
@m1entus Actually `-layoutDocumentView` is just a method I made that updates the document view frame based on the size calculated in the layout. You'd want to recalculate the layout before updating the document view frame.